docs/advanced/extensibleenums.md

```diff
@@ -7,216 +7,134 @@ Extensible Enums are an enhancement of specific Vanilla enums to allow new entri
 
-    // ...
-}
-```
+- If the parameter is an int ID parameter related to a `@IndexedEnum` annotation on the enum, then it is ignored and replaced by the entry's ordinal. If said parameter is specified inline in the JSON, then it must be specified as `-1`
+- If the parameter is a String name parameter related to a `@NamedEnum` annotation on the enum, then it must be prefixed by the mod ID in the `namespace:path` format known from `ResourceLocation`s
 
-For `StringRepresentable` enums, `IExtensibleEnum` provides `createCodecForExtensibleEnum` and `createStreamCodecForExtensibleEnum` which do not cache the values at all, meaning they will always obtain up-to-date enums added at any point.
+#### Retrieving the generated constant
 
-```java
-// For some enum with codec
-public enum ExampleEnumStringCodec implements StringRepresentable {
-    // ...
-    ;
+The generated enum constant can be retrieved via `TheEnum.valueOf(String)`. If a field reference is used to provide the parameters, then the constant can also be retrieved from the `EnumProxy` object via `EnumProxy#getValue()`
 
-    private static final IntFunction<ExampleEnumStringCodec> BY_ID = ByIdMap.continuous(
-        ExampleEnumStringCodec::getId, values(), ByIdMap.OutOfBoundsStrategy.ZERO
-    );
-
-    public static final Codec<ExampleEnumStringCodec> CODEC = StringRepresentable.fromEnum(ExampleEnumStringCodec::values);
-    public static final StreamCodec<ByteBuf, ExampleEnumStringCodec> CODEC = ByteBufCodecs.idMapper(BY_ID, ExampleEnumStringCodec::getId);
+## Contributing to NeoForge
 
-    ExampleEnumStringCodec(int id, String serializedName) {
-        // ...
-    }
+To add a new extensible enum to NeoForge, there are at least two required things to do:
 
-    // ...
-}
+- First, have the enum implement `IExtensibleEnum` to mark that this enum should be transformed via the `RuntimeEnumExtender`
+- Second, add a `getExtensionInfo` method which returns `ExtensionInfo.nonExtended(TheEnum.class)`
 
-// The final extended enum may look something like this
-public enum ExampleEnumStringCodec implements StringRepresentable, net.neoforged.neoforge.common.IExtensibleEnum {
-    // ...
-    ;
+Further action is required depending on specific details about the enum:
 
-    private static final java.util.Map<String, ExampleEnumStringCodec> BY_NAME = 
-        java.util.Arrays.stream(ExampleEnumStringCodec.values())
-        .collect(java.util.stream.Collectors.toMap(
-            e -> e.serializedName, e -> e
-        ));
+- If the enum has an int ID parameter which should match the entry's ordinal, then the enum should be annotated with `@NumberedEnum` with the ID's parameter index as the annotation's value if it's not the first parameter
+- If the enum has a String name parameter which is used for serialization and should therefore be namespaced, then the enum should be annotated with `@NamedEnum` with the name's parameter index as the annotation's value if it's not the first parameter
```

Comment:
I am confused as to what the annotation parameters are supposed to be, needs improved wording.

Reply:
The annotation parameter is the relevant constructor parameter's index. In most cases that would be 0 (the annotation's default value), except when the given constructor parameter is not the first one (i.e. when an enum has an ID and a name parameter as the first and second parameter respectively, then index 1 needs to be specified in the annotation for the latter). Not quite sure at the moment how to word that better.